label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleValidatorImpl.java

```diff
@@ -123,31 +122,55 @@ private void validateJsonCondition(JsonCondition jsonCondition) {
     }
   }
 
+  private void validateKeyStringCondition(StringCondition stringCondition) {
```

Comment:
shouldn't this be validating the value in addition to the operator? In general, why does it need a different validator than the general string condition?

Reply:
Yes missed validating the value. It needs a different validator because we support only equals and regex match for key condition. Thus, adding separate validations for it. Ideally we should have used a different Operator enum for key condition.

Reply:
Got it

>  Ideally we should have used a different Operator enum for key condition.

Probably other way around :) key condition was here before the ip types were recently added.